lib/SIL/SILVerifier.cpp

```diff
@@ -577,6 +578,38 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
     SILLocation::LocationKind LocKind = L.getKind();
     ValueKind InstKind = I->getKind();
 
+    // Check that there is at most one debug variable defined
+    // for each argument slot.
```

Comment:
... This catches SIL transformations that accidentally remove inline information (stored in the SILDebugScope) form debug-variable-carrying instructions.